src/components/Layout/Header/ActionCenter/components/Details/LimitOrderDetails.tsx

```diff
@@ -1,49 +1,170 @@
+    if (!order) return undefined
+    if (!isLimitOrderAction(action)) return undefined
+
+    if (order.status !== OrderStatus.OPEN) return undefined
+
+    return {
+      accountId: action.limitOrderMetadata.accountId,
+      sellAssetId: action.limitOrderMetadata.sellAsset.assetId,
+      buyAssetId: action.limitOrderMetadata.buyAsset.assetId,
+      order,
+    }
+  }, [action, order])
+
+  const handleCancelOrder = useCallback(() => {
+    if (!orderToCancel) return
+    onCancelOrder(orderToCancel)
+  }, [onCancelOrder, orderToCancel])
 
   return (
     <Stack gap={4}>
       <Row fontSize='sm'>
         <Row.Label>{translate('notificationCenter.pair')}</Row.Label>
         <Row.Value>
-          <RawText>ETH/USDC</RawText>
+          <RawText>{pair}</RawText>
         </Row.Value>
       </Row>
       <Row fontSize='sm'>
         <Row.Label>{translate('notificationCenter.limitPrice')}</Row.Label>
         <Row.Value>
-          <RawText>1 ETH = 0.00 USDC</RawText>
-        </Row.Value>
-      </Row>
-      <Row fontSize='sm'>
-        <Row.Label>{translate('notificationCenter.expires')}</Row.Label>
-        <Row.Value>
-          <RawText>10 minutes</RawText>
-        </Row.Value>
-      </Row>
-      <Row fontSize='sm'>
-        <Row.Label>{translate('notificationCenter.executionPrice')}</Row.Label>
-        <Row.Value>
-          <RawText>1 ETH = 0.00 USDC</RawText>
+          <HStack>
+            <Amount.Crypto value='1.0' symbol={sellAsset?.symbol ?? ''} />
+            <RawText>=</RawText>
+            <Amount.Crypto
+              value={limitPrice?.buyAssetDenomination}
+              symbol={buyAsset?.symbol ?? ''}
+            />
+          </HStack>
         </Row.Value>
       </Row>
+      {expires ? (
+        <Row fontSize='sm'>
+          <Row.Label>{translate('notificationCenter.expires')}</Row.Label>
+          <Row.Value>
+            <TransactionDate blockTime={expires} />
+          </Row.Value>
+        </Row>
+      ) : null}
+      {executedBuyAmountCryptoBaseUnit && executedSellAmountCryptoBaseUnit ? (
+        <Row fontSize='sm'>
+          <Row.Label>{translate('notificationCenter.executionPrice')}</Row.Label>
+          <Row.Value>
+            <HoverTooltip placement='top' label={executionPriceCryptoFormatted}>
+              <Amount.Crypto
+                value={executionPrice}
+                symbol={buyAsset?.symbol ?? ''}
+                maximumFractionDigits={6}
+              />
+            </HoverTooltip>
+          </Row.Value>
+        </Row>
+      ) : null}
       <Row fontSize='sm'>
         <Row.Label>{translate('notificationCenter.filled')}</Row.Label>
         <Row.Value display='flex' alignItems='center' gap={2}>
-          <Progress width='100px' size='xs' value={50} colorScheme='green' />
-          <RawText>50%</RawText>
+          <Progress
+            width='100px'
+            size='xs'
+            value={bnOrZero(filledDecimalPercentage).times(100).toNumber()}
```

Comment:
q: should we also have this exposed as percentage human (or whatever we call this) instead of inlining ?

Reply:
Yeah make sense, did in [cdf0b44](https://github.com/shapeshift/web/pull/9626/commits/cdf0b44a2fa6ccc6a07536159b35c86d1f1a1a26)